modules/streams/src/main/scala/dev/profunktor/redis4cats/streams/streams.scala

```diff
@@ -49,20 +50,17 @@ trait Streaming[F[_], K, V] {
     *
     * Note that if you block indefinitely or longer than the configured timeout for the underlying Lettuce client,
     * Lettuce will terminate the stream with `io.lettuce.core.RedisCommandTimeoutException`. To avoid this set
-    * `restartOnTimeout` to `Some`, but then your stream will not be aware of any connection issues that silently
-    * stop sending data.
+    * `restartOnTimeout` to `RestartOnTimeout.always`, but then your stream will not be aware of any connection
+    * issues that silently stop sending data.
     *
     * @see https://redis.io/commands/xread
-    *
-    * @param restartOnTimeout if `Some`, receives elapsed time since the stream started and determines whether to
-    *                         restart the stream based on the returned boolean (true to restart).
     */
   def read(
       keys: Set[K],
       chunkSize: Int,
       initialOffset: K => StreamingOffset[K] = StreamingOffset.All[K],
       block: Option[Duration] = Some(Duration.Zero),
       count: Option[Long] = None,
-      restartOnTimeout: Option[FiniteDuration => Boolean] = None
+      restartOnTimeout: RestartOnTimeout = RestartOnTimeout.never
```

Comment:
I think this should be ```.always``` .
Keep in mind a RedisCommandTimeoutException is unrelated to network issues, rather its simpoly how long a command takes to run. When running a blocking command that waits for data, it will wait until there is data

Reply:
But I think it is related to network issues. If the network goes dark without any traffic on the line, you can hang there indefinitely, IIRC. Timeouts resolve that situation.

Maybe we should not provide a default here at all and force the user to choose an option.

Reply:
There is another exception that is thrown for connection issues. This exception is not about network issues